src/leap/mail/imap/mailbox.py

```diff
@@ -70,33 +73,29 @@ def do_profile_cmd(d, name):
         d.addCallback(_debugProfiling, name, time.time())
         d.addErrback(lambda f: log.msg(f.getTraceback()))
 
+INIT_FLAGS = (MessageFlags.SEEN_FLAG, MessageFlags.ANSWERED_FLAG,
+              MessageFlags.FLAGGED_FLAG, MessageFlags.DELETED_FLAG,
+              MessageFlags.DRAFT_FLAG, MessageFlags.RECENT_FLAG,
+              MessageFlags.LIST_FLAG)
 
-class SoledadMailbox(WithMsgFields, MBoxParser):
+
+class IMAPMailbox(object):
     """
     A Soledad-backed IMAP mailbox.
 
     Implements the high-level method needed for the Mailbox interfaces.
-    The low-level database methods are contained in MessageCollection class,
-    which we instantiate and make accessible in the `messages` attribute.
+    The low-level database methods are contained in IMAPMessageCollection
+    class, which we instantiate and make accessible in the `messages`
+    attribute.
     """
     implements(
         imap4.IMailbox,
         imap4.IMailboxInfo,
-        imap4.ICloseableMailbox,
+        #imap4.ICloseableMailbox,
```

Comment:
I recently discovered that I had origininally implemented this interface but it might not be needed at all, since it seems to be optional and I don't think we need any extra cleanup other than what's done at expunge. But I want to be sure i'm not missing something, I'll leave a comment for now.